docs/user-guide/advanced_topics/dynamic_federations.md

```diff
@@ -0,0 +1,46 @@
+# Dynamic Federations
+
+In general a dynamic federation is one in which the federates or simulators are not all available at the start of the co-simulation. But instead some components attach to the running co-simulation part-way through the simulation. The longer term goal of HELICS is to fully support dynamic federations but there are many complicated issues to handle timing and data communication that need to be resolved before full general dynamic federations are supported. As of HELICS 3.1 a limited though quite useful form is available through observer federates.
+
+## Levels of Dynamic Federations
+
+Dynamic Federations come in various flavors of complexity. The first is simply allowing additional subscriptions to existing federates. The next is allowing additional "observer" federates to a running co-simulation. The third is allowing new publications, endpoints, or filters on existing federates. And the fourth is a full dynamic federation.
+
+The first of these has been unofficially available for some time and will be supported in HELICS 3.1. Dynamic observer federates will also be allowed in HELICS 3.1, and will be detailed further in a subsequent section. The last two levels are currently a work in progress and are currently planned to be supported n HELICS 3.2 (Early 2022).
+
+### Dynamic Subscriptions
+
+In the normal case, the registerInput, registerSubscription, or registerTargettedEndpoint are done in the creation phase of co-simulation so all the checks and connections can be made and everything set up before time Zero (start of co-simulation). If these calls are made after the enterInitialization call, then the time guarantees are different. Ordinarily data published in the initialization phase would be available to all subscriptions of a publication, But with dynamic subscriptions the data is not going to be sent to the new subscription until new publications after a timeRequest or EnterExecutingMode call has been made. ( A bufferData flag is in the works to allow data to be sent for dynamic subscriptions at the time of connection if set on the publisher).
+
+Apart from time considerations there are no additional requirements. The checks on the subscription are done immediately as opposed to waiting for the `enterInitializationMode` call.
+
+### Dynamic Observer Federates
+
+A federate may declare itself to be and observer in the FederateInfo Structure when a federate is declared. This can be done via command line (`--observer`) or through a flag.
```

Comment:
```suggestion
A federate may declare itself to be an observer in the FederateInfo structure when a federate is declared. This can be done via the command line (`--observer`) or through a flag.
```